OPAL/br/src/main/scala/org/opalj/br/analyses/cg/EntryPointFinder.scala

```diff
@@ -357,55 +359,37 @@ object AllEntryPointsFinder extends EntryPointFinder {
  */
 object AndroidEntryPointsFinder extends EntryPointFinder {
 
-    val activityEntryPoints: List[(String, Option[MethodDescriptor])] = List(
-        ("onCreate", Some(MethodDescriptor.apply(FieldType("Landroid/os/Bundle;"), VoidType))),
-        ("onRestart", Some(MethodDescriptor.NoArgsAndReturnVoid)),
-        ("onStart", Some(MethodDescriptor.NoArgsAndReturnVoid)),
-        ("onResume", Some(MethodDescriptor.NoArgsAndReturnVoid)),
-        ("onStop", Some(MethodDescriptor.NoArgsAndReturnVoid)),
-        ("onDestroy", Some(MethodDescriptor.NoArgsAndReturnVoid)),
-        ("onActivityResult", Some(MethodDescriptor.NoArgsAndReturnVoid))
-    )
-    val serviceEntryPoints: List[(String, Option[MethodDescriptor])] = List(
-        ("onCreate", Some(MethodDescriptor.NoArgsAndReturnVoid)),
-        ("onStartCommand", Some(MethodDescriptor.apply(FieldTypes(FieldType("Landroid/content/Intent;"), IntegerType, IntegerType), VoidType))),
-        ("onBind", Some(MethodDescriptor.apply(FieldType("Landroid/content/Intent;"), VoidType))),
-        ("onRebind", Some(MethodDescriptor.apply(FieldType("Landroid/content/Intent;"), VoidType))),
-        ("onStart", Some(MethodDescriptor.apply(FieldTypes(FieldType("Landroid/content/Intent;"), IntegerType), VoidType))),
-        ("onDestroy", Some(MethodDescriptor.NoArgsAndReturnVoid))
-    )
-    val contentProviderEntryPoints: List[(String, Option[MethodDescriptor])] = List(
-        ("onCreate", Some(MethodDescriptor.NoArgsAndReturnVoid)),
-        ("query", None),
-        ("insert", None),
-        ("update", None)
-    )
-    val locationListenerEntryPoints: List[(String, Option[MethodDescriptor])] = List(
-        ("onLocationChanged", None),
-        ("onProviderDisabled", Some(MethodDescriptor.JustTakes(FieldType("Ljava/lang/String;")))),
-        ("onProviderEnabled", Some(MethodDescriptor.JustTakes(FieldType("Ljava/lang/String;")))),
-        ("onStatusChanged", Some(MethodDescriptor.apply(FieldTypes(FieldType("Ljava/lang/String;"), IntegerType, FieldType("Landroid/os/Bundle;")), VoidType)))
-    )
-    val broadcastReceiverEntryPoints: List[(String, Option[MethodDescriptor])] = List(
-        ("onReceive", Some(MethodDescriptor.apply(FieldTypes(FieldType("Landroid/content/Context;"), FieldType("Landroid/content/Intent;")), VoidType)))
-    )
-    val defaultEntryPoints = Map(
-        "android/app/Activity" -> activityEntryPoints,
-        "android/app/Service" -> serviceEntryPoints,
-        "android/content/ContentProvider" -> contentProviderEntryPoints,
-        "android/location/LocationListener" -> locationListenerEntryPoints,
-        "android/content/BroadcastReceiver" -> broadcastReceiverEntryPoints
-    )
-
     override def collectEntryPoints(project: SomeProject): Traversable[Method] = {
         val entryPoints = ArrayBuffer.empty[Method]
+
+        val defaultEntryPoints = Map(
```

Comment:
Better, but I'm not sure an immutable map with mutable ListBuffers is a good idea. You should be able to just iterate over the keys in the config entry (and if that is not possible, start with a list of the keys here and create the map on the fly when parsing the config to create a fully immutable map.